integration-tests/src/test_loop/utils/resharding.rs

```diff
@@ -80,7 +83,7 @@ pub(crate) fn fork_before_resharding_block(double_signing: bool) -> LoopAction {
                         base_block_height: tip.height - 1,
                     }
                 };
-                client_actor.adv_produce_blocks_on(3, true, height_selection);
+                client_actor.adv_produce_blocks_on(blocks_produced, true, height_selection);
```

Comment:
Why move it to an argument? 

What exactly does this control? Does it have to be 3 in order to trigger fork at the right place?

Reply:
it's `num of block produced`

Yes it's a way to change which fork gets finalized. Small number of produced blocks -> discarded, large number -> not discarded